command/agent/health_endpoint.go

```diff
@@ -148,7 +149,22 @@ func (s *HTTPServer) HealthServiceNodes(resp http.ResponseWriter, req *http.Requ
 
 	// Filter to only passing if specified
 	if _, ok := params[api.HealthPassing]; ok {
-		out.Nodes = filterNonPassing(out.Nodes)
+		val := params.Get(api.HealthPassing)
```

Comment:
I had that originally, but it looks like this after you fill in the `parse the bool` comment:

```golang
var filter bool
val := params.Get(api.HealthPassing)
if val == "" {
    filter = true
} else {
    filt, err := strconv.ParseBool(val)
    if err != nil {
      resp.WriteHeader(400)
      fmt.Fprint(resp, "error")
      return nil, nil
    }
    filter = filt
}
if filter {
    out.Nodes = filterNonPassing(out.Nodes)
}
```

or you shadow filter, which felt hacky